### DISCUSSION TITLE
Some questions  about Minio's  Shared link

### DISCUSSION THREAD

**Discussion by yanghui-git** [minio/minio#12346] [Q&A] [answered]
I have three questions,  are there some solutions？

The Background is  I will share the download link with someone  like I will shard test1.txt  download link with tom.

First, Control the number of downloads ,for example  the shard link  only can  access three times 

Second ,Control the effective duration of the link like in max 5 seconds  ,otherwise give up,

Third ,Someone  can access the shard link and download  limited by  bucket I set  like bucket test1 ,test2

I am sorry My English is construction site level ，hahahaha

Thanks very much 

In Chinese is
背景 :基于minio的分享链接操作，完成附加功能。
1.  控制下载次数，minio的链接只能下载x次
2. 控制链接有效时长，精确到秒
3. 控制下载权限，只有某些来源能下载

**Comment by harshavardhana** [comment, marked as answer]
Shared link is speced on AWS S3 presigned URL. AFAIK there is no way to limit the number of downloads etc.

This needs to be managed at a layer above MinIO.

**Reply by yanghui-git** [reply]
So  can i do this by using gateway component like nginx  or Integration permissions like  springsecurity  auth2 or others
